src/Kiota.Builder/Refiners/CommonLanguageRefiner.cs

```diff
@@ -12,20 +12,61 @@ public abstract class CommonLanguageRefiner : ILanguageRefiner
             _configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
         }
         public abstract void Refine(CodeNamespace generatedCode);
+        protected void AddSerializationModulesImport(CodeElement generatedCode) {
+            if(generatedCode is CodeMethod currentMethod &&
+                currentMethod.IsOfKind(CodeMethodKind.ClientConstructor) &&
+                currentMethod.Parent is CodeClass currentClass &&
+                currentClass.StartBlock is CodeClass.Declaration declaration) {
+                    var cumulatedSymbols = currentMethod.DeserializerModules.Union(currentMethod.SerializerModules).ToList();
+                    currentMethod.DeserializerModules = currentMethod.DeserializerModules.Select(x => x.Split('.').Last()).ToList();
+                    currentMethod.SerializerModules = currentMethod.SerializerModules.Select(x => x.Split('.').Last()).ToList();
+                    declaration.Usings.AddRange(cumulatedSymbols.Select(x => new CodeUsing(currentClass){
+                        Name = x.Split('.').Last(),
+                        Declaration = new CodeType(currentClass) {
+                            Name = x.Split('.').SkipLast(1).Aggregate((x, y) => $"{x}.{y}"),
+                            IsExternal = true,
+                        }
+                    }));
+                    return;
+                }
+            CrawlTree(generatedCode, AddSerializationModulesImport);
+        }
+        public void ReplaceDefaultSerializationModules(CodeElement generatedCode, params string[] moduleNames) {
+            if(generatedCode is CodeMethod currentMethod &&
+                currentMethod.IsOfKind(CodeMethodKind.ClientConstructor) &&
+                currentMethod.SerializerModules.Count == 1 &&
+                currentMethod.SerializerModules.Any(x => "Microsoft.Kiota.Serialization.Json.JsonSerializationWriterFactory".Equals(x, StringComparison.OrdinalIgnoreCase))) {
+                currentMethod.SerializerModules.Clear();
+                currentMethod.SerializerModules.AddRange(moduleNames);
+                return;
+            }
+            CrawlTree(generatedCode, (x) => ReplaceDefaultSerializationModules(x, moduleNames));
+        }
+        public void ReplaceDefaultDeserializationModules(CodeElement generatedCode, params string[] moduleNames) {
+            if(generatedCode is CodeMethod currentMethod &&
+                currentMethod.IsOfKind(CodeMethodKind.ClientConstructor) &&
+                currentMethod.DeserializerModules.Count == 1 &&
+                currentMethod.DeserializerModules.Any(x => "Microsoft.Kiota.Serialization.Json.JsonParseNodeFactory".Equals(x, StringComparison.OrdinalIgnoreCase))) {
+                currentMethod.DeserializerModules.Clear();
+                currentMethod.DeserializerModules.AddRange(moduleNames);
+                return;
+            }
```

Comment:
These two methods have the same code except for 1 string. Consider factoring this out into a single method.